vllm/platforms/cuda.py

```diff
@@ -608,6 +608,66 @@ def log_warnings(cls):
                     ", ".join(device_names),
                 )
 
+    @classmethod
+    @with_nvml_context
+    def set_cpu_affinity(cls, device_id: int) -> None:
+        """
+        Set CPU affinity for the current process based on GPU device ID.
+        """
+        try:
+            import psutil
+        except ImportError:
+            logger.warning(
+                "psutil is not available. Cannot set CPU affinity. "
+                "Install psutil to enable NUMA affinity optimization.")
+            return
+
+        try:
+            physical_device_id = cls.device_id_to_physical_device_id(device_id)
+            handle = pynvml.nvmlDeviceGetHandleByIndex(physical_device_id)
+
+            # Get CPU affinity for this GPU
+            # We need to determine the CPU set size first
+            cpu_count = os.cpu_count()
```

Comment:
Should we get physical core count only or virtual core count?   `os.cpu_count()` will get virtual core count.   For physical core count `psutil.cpu_count(logical=False)` can be used.